### PR TITLE
feat(accordion): add accordion bem story + design tokens

### PR DESCRIPTION
**Pull request by robbieatacato**
### Solving issue:
https://github.com/nl-design-system/denhaag/issues/576

### Change:

- add accordion html (based off Mui, cleaned out Mui classes) + css + design tokens
- accordion default
- accordion expanded
- accordion disabled

**Closing issues**

### CODE REVIEW

File: proprietary/Components/src/denhaag/accordion.tokens.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "denhaag": {
+    "accordion": {
+      "border": { "value": "1px solid {denhaag.color.grey.2.value}" },
+      "font-family": { "value": "{denhaag.typography.sans-serif.font-family.value}" },
+      "font-weight": { "value": "400" },
+      "line-height": { "value": "24px" },
+      "padding": { "value": "16px" },
+      "margin": { "value": "12px" },
+      "expanded": {
+        "font-weight": { "value": "500" },
```
Comment (paulAcato) [inline proprietary/Components/src/denhaag/accordion.tokens.json:11]: Today i learned when the font shows like this in Figma,
Screenshot 2021-12-09 at 12 31 02

The font-weight is 700.

**Review by paulAcato — CHANGES_REQUESTED** [review]

File: components/Accordion/src/accordion.css [hunk 2]
```diff
@@ -0,0 +1,122 @@
+/* accordion general styling */
+.denhaag-accordion {
+  font-family: var(--denhaag-accordion-font-family), var(--denhaag-typography-sans-serif-font-family);
```
Comment (Robbert) [inline components/Accordion/src/accordion.css:3]: Fallback need to be defined slightly differently:

`var(--a), var(--b)` → `var(--a, var(--b))`

File: components/Accordion/src/stories/bem.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                class="denhaag-accordion__panel denhaag-accordion__panel--disabled"
+                tabindex="0"
+                role="button"
+                aria-disabled="true"
+                aria-expanded="false"
+              >
+                <div class="denhaag-accordion__summary">
+                  <p class="denhaag-accordion__title">Accordion</p>
+                </div>
+                <div class="" aria-disabled="false" aria-hidden="true">
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx:184]: The parent is not disabled: 
```
<div
                class="denhaag-accordion__panel"
                tabindex="0"
                role="button"
                aria-disabled="**_false_**"
                aria-expanded="false"
              >
```

[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx:184]: Klopt

File: components/Accordion/src/accordion.css (hunk 2 above)
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+.denhaag-accordion-container {
+  border: var(--denhaag-accordion-border);
+  position: relative;
+}
+
+.denhaag-accordion-container:last-child {
+  border-bottom-left-radius: var(--denhaag-border-radius);
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  border-top-right-radius: var(--denhaag-border-radius);
+  border-top: var(--denhaag-accordion-border);
+}
+
+/* panel (summary) styling */
+.denhaag-accordion__panel {
+  color: inherit;
+  border: 0;
+  cursor: pointer;
+  margin-block-start: 0;
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  border: var(--denhaag-accordion-focus-border);
+}
+
+.denhaag-accordion__summary {
+  margin-block-start: var(--denhaag-accordion-margin);
+  margin-block-end: var(--denhaag-accordion-margin);
+  flex-grow: 1;
+  padding-inline-end: var(--denhaag-accordion-padding);
+  font-weight: var(--denhaag-accordion-font-weight);
+  color: var(--denhaag-color-grey-4);
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+.denhaag-accordion__title {
+  font-style: normal;
+  font-size: var(--denhaag-typography-scale-base-font-size);
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+
+## Design Tokens
+
+<DesignTokenDocBlock categoryName="Accordion" viewType="table" />
+
+## Stories
+
+### Default
+
+<Canvas>
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx:33]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/fd9aef5dc3b04bbe13cc97e607daa3108ebea8a3

File: components/Accordion/src/accordion.css (hunk 2 above)
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: proprietary/Components/src/denhaag/accordion.tokens.json (hunk 1 above)
Comment (robbieatacato) [reply proprietary/Components/src/denhaag/accordion.tokens.json]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+                      aria-hidden="true"
+                      shape-rendering="auto"
+                    >
+                      <path
+                        d="M5.293 8.293a1 1 0 011.414 0L12 13.586l5.293-5.293a1 1 0 111.414 1.414l-6 6a1 1 0 01-1.414 0l-6-6a1 1 0 010-1.414z"
+                        fill="currentColor"
+                      ></path>
+                    </svg>
+                  </span>
+                  <span class=""></span>
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+                </div>
+              </div>
+              <div class="denhaag-accordion__collapse">
+                <div class="">
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+  line-height: var(--denhaag-accordion-line-height);
+  display: flex;
+  align-items: center;
+  margin-block-start: 0;
+  margin-block-end: 0;
+}
+
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,227 @@
...
+  {/* eslint-disable react/no-unknown-property */}
+  <Story name="Disabled">
+    <div>
+      <div class="story--components-surfaces-accordion--disabled">
+        <div class="denhaag-theme">
+          <div class="denhaag-accordion">
+            <div class="denhaag-accordion-container denhaag-accordion-container--disabled">
+              <div
+                class="denhaag-accordion__panel denhaag-accordion__panel--disabled"
+                tabindex="0"
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+.denhaag-accordion__icon--expanded .denhaag-icon {
+  transform: var(--denhaag-accordion-expanded-transform);
+  color: var(--denhaag-color-blue-3);
+}
+
+.denhaag-accordion__summary--expanded,
+.denhaag-accordion__panel:not(.denhaag-accordion__panel--disabled):hover .denhaag-accordion__title,
+.denhaag-accordion__panel:not(.denhaag-accordion__panel--disabled):hover .denhaag-icon {
+  color: var(--denhaag-color-blue-3);
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/accordion.css [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+.denhaag-accordion-container:first-child {
+  border-top-left-radius: var(--denhaag-border-radius);
+  border-top-right-radius: var(--denhaag-border-radius);
+  border-top: var(--denhaag-accordion-border);
+}
+
+/* panel (summary) styling */
+.denhaag-accordion__panel {
+  color: inherit;
+  border: 0;
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/accordion.css]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

File: components/Accordion/src/stories/bem.stories.mdx [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+  }}
+/>
+
+<Description markdown={readme} />
+
+## Design Tokens
+
+<DesignTokenDocBlock categoryName="Accordion" viewType="table" />
+
+## Stories
```
[the remark this answers is not in the document]
Comment (robbieatacato) [reply components/Accordion/src/stories/bem.stories.mdx:29]: Updated in commit: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25

**Comment by robbieatacato** [comment]
@paulAcato @Robbert I did a 

- refactor of the Bem story, css and tokens: https://github.com/nl-design-system/denhaag/pull/600/commits/1ae898c3f293a8c5c8e5862c65c9f3ec4c1e8d25
- added a functional component variant of ModalScripts but for the accordion: https://github.com/nl-design-system/denhaag/pull/600/commits/fd9aef5dc3b04bbe13cc97e607daa3108ebea8a3

**Comment by paulAcato** [comment]
@robbieatacato , can you make sure your builds don't fail? https://github.com/nl-design-system/denhaag/runs/4533850487?check_suite_focus=true#step:7:97

**Comment by paulAcato** [comment]
@robbieatacato , I'm wondering, why did u merge the breadcrumbs component (which is still in PR) into this branch?

**Comment by bddjong** [comment]
This PR contains a lot of changes which aren't a supposed to be a part of it. It's very difficult to properly review, and cant be merged until that's fixed

**Pull request closed without merging**